flow/Arena.cpp

```diff
@@ -216,15 +220,23 @@ size_t ArenaBlock::totalSize() const {
 	while (o) {
 		ArenaBlockRef* r = (ArenaBlockRef*)((char*)getData() + o);
 		makeDefined(r, sizeof(ArenaBlockRef));
-		if (r->aligned4kBufferSize != 0) {
-			totalSizeEstimate += r->aligned4kBufferSize;
+		o = r->nextBlockOffset;
+		if (r->type == ArenaBlockRef::Type::kAligned4kBuffer) {
+			makeDefined(r, sizeof(Aligned4kBufferRef));
+			auto* b = static_cast<Aligned4kBufferRef*>(r);
+			totalSizeEstimate += b->aligned4kBufferSize;
+			makeNoAccess(r, sizeof(Aligned4kBufferRef));
+		} else if (r->type == ArenaBlockRef::Type::kResourceWithDestructor) {
+			// No size estimate available
```

Comment:
I don't know how important this is, but it's possible we could provide a size estimate as an argument to the function adding the resource.